.github/workflows/on_pull_request.yaml

```
@@ -0,0 +1,18 @@
+name: On Pull Request
+
+on:
+  pull_request:
+
+jobs:
+
+  on-pull-request:
+    name: Get rocks modified and build-scan-test them
+    uses: canonical/charmed-kubeflow-workflows/.github/workflows/get-rocks-modified-and-build-scan-test-publish.yaml@main
+    permissions:
+      pull-requests: read
+    secrets: inherit
+    with:
+      microk8s-channel: 1.32-strict/stable
+      juju-channel: 3.6/stable
+      python-version: "3.9"
```

question: Not sure why we use Python `3.9` here. The `resource-dispatcher` Dockerfile uses `3.10`, and the `rockcraft.yaml` here uses Ubuntu `22.04` as a base which also has Python `3.10` as default.

Though from what I understand, this input is [only used](https://github.com/canonical/charmed-kubeflow-workflows/blob/8cab1ffdfecf199734e10df7e805de6aa42987a5/.github/workflows/build-scan-test-publish-rock.yaml#L66) for the [integration-test-rock](https://github.com/canonical/charmed-kubeflow-workflows/blob/8cab1ffdfecf199734e10df7e805de6aa42987a5/.github/workflows/integration-test-rock.yaml) action, and we don't have any integration tests so it doesn't matter
```suggestion
      python-version: "3.9"
```

---

sorry I have fixed it and set the python to 3.10 because resource dispatcher uses python dependencies built for 3.10 https://github.com/canonical/resource-dispatcher-image/blob/main/requirements.txt#L2